dev/integration_tests/abstract_method_smoke_test/android/app/src/main/kotlin/com/example/abstract_method_smoke_test/MainActivity.kt

```diff
@@ -1,35 +1,61 @@
 package com.example.abstract_method_smoke_test
 
 import android.content.Context
+import android.graphics.Color
 import android.os.Bundle
+import android.view.View
 import android.view.inputmethod.InputMethodManager
 import io.flutter.app.FlutterActivity
-import io.flutter.plugin.common.MethodChannel;
-import io.flutter.plugin.common.MethodChannel.MethodCallHandler;
+import io.flutter.plugin.common.MethodChannel
+import io.flutter.plugin.common.StandardMessageCodec
+import io.flutter.plugin.platform.PlatformView
+import io.flutter.plugin.platform.PlatformViewFactory
 import io.flutter.plugins.GeneratedPluginRegistrant
 
-class MainActivity: FlutterActivity() {
-  override fun onCreate(savedInstanceState: Bundle?) {
-    super.onCreate(savedInstanceState)
-    GeneratedPluginRegistrant.registerWith(this)
-
-    // Triggers the Android keyboard, which causes the resize of the Flutter view.
-    // We need to wait for the app to complete.
-    MethodChannel(getFlutterView(), "com.example.abstract_method_smoke_test")
-        .setMethodCallHandler({ call, result ->
-          toggleInput()
-          result.success(null)
-        })
-  }
-
-  override fun onPause() {
-    // Hide the input when the app is closed.
-    toggleInput()
-    super.onPause()
-  }
-
-  fun toggleInput() {
-    val imm = getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
-    imm.toggleSoftInput(InputMethodManager.SHOW_IMPLICIT, 0)
-  }
+class MainActivity : FlutterActivity() {
+    class SimplePlatformView(context: Context) : PlatformView {
+        private val view: View = View(context)
+
+        init {
+            view.setBackgroundColor(Color.CYAN)
+        }
+
+        override fun dispose() {}
```

Comment:
not sure if the empty `dispose()` is needed, but just in case.

Reply:
Hard to tell in Kotlin, but the original class is a Java interface so it's still needed here.